common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -0,0 +1,261 @@
+        final Map<String, String> parameters = StringExtensions.getUrlParameters(url);
+        if (!StringExtensions.isNullOrBlank(parameters.get("error"))) {
+            Logger.info(TAG, "Sending intent to cancel authentication activity");
+            Intent resultIntent = new Intent();
+            resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_CODE, parameters.get("error"));
+            resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE, parameters.get("error_description"));
+            getCompletionCallback().sendResponse(AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL, resultIntent);
+            view.stopLoading();
+        } else {
+            Logger.verbose(TAG, "It is pointing to redirect. Final url can be processed to get the code or error.");
+            Intent resultIntent = new Intent();
+            resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_FINAL_URL, url);
+            resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_REQUEST_INFO,
+                    getRequest());
+            getCompletionCallback().sendResponse(
+                    AuthenticationConstants.UIResponse.BROWSER_CODE_COMPLETE,
+                    resultIntent);
+            view.stopLoading();
+            //the TokenTask should be processed at after the authorization process in the upper calling layer.
+        }
+
+        return true;
+    }
+
+    private boolean processWebsiteRequest(@NonNull final WebView view, @NonNull final String url) {
+        //Open url link in browser
+        final String link = url
+                .replace(AuthenticationConstants.Broker.BROWSER_EXT_PREFIX, "https://");
+        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(link));
+        getActivity().getApplicationContext().startActivity(intent);
+        view.stopLoading();
+        Intent resultIntent = new Intent();
+        getCompletionCallback().sendResponse(AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL, resultIntent);
+        return true;
+    }
+
+    private boolean processInstallRequest(@NonNull final WebView view, @NonNull final String url) {
+        Logger.verbose(TAG, "Return to caller with BROKER_REQUEST_RESUME, and waiting for result.");
+        final Intent resultIntent = new Intent();
+        getCompletionCallback().sendResponse(AuthenticationConstants.UIResponse.BROKER_REQUEST_RESUME, resultIntent);
+
+        // Having thread sleep for 1 second for calling activity to receive the result from
+        // prepareForBrokerResumeRequest, thus the receiver for listening broker result return
+        // can be registered. openLinkInBrowser will launch activity for going to
+        // playstore and broker app download page which brought the calling activity down
+        // in the activity stack.
+        final int threadSleepForCallingActivity = 1000;
+        try {
+            Thread.sleep(threadSleepForCallingActivity);
```

Comment:
Usually, we should avoid using 'Thread.sleep' since it might cause ANR. Could we try some other methods, like "Handler.postDelayed"?